src/awesome_map/BoundaryInterceptor.js

```diff
@@ -232,10 +322,18 @@ define(function(require) {
                 this._touchState = targetState;
                 break;
 
-            case EventTypes.DRAG:
             case EventTypes.DRAG_END:
+                /* jshint -W086 */// Expected break statement 
+
+                this._debouncedDetermineBoundaryVisibility();
+                // Fall through
+
+            case EventTypes.DRAG:
             case EventTypes.DRAG_START:
+                /* jshint +W086 */// Expected break statement             
 
+                this._updateBoundariesStillVisible();
```

Comment:
I admit that I am having a little bit of trouble following this.  You get a new visibility list for drag ends and update that, but only update drag and drag starts?  It probably works, but an elucidating comment might help the abstract-challenged.